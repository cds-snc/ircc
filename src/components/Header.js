import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Title from './Title'

const Div = styled.div`
  color: #fff;
  background-color: #2572b4;
  border-color: #2572b4;
  padding: 10px 15px;
  border-bottom: 1px solid transparent;
  border-top-right-radius: 3px;
  border-top-left-radius: 3px;
`

const Header = ({ title }) =>
  <Div>
    <Title>
      {title}
    </Title>
  </Div>

Header.propTypes = {
  title: PropTypes.string,
}

export default Header
