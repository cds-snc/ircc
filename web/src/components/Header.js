import React from 'react'
import PropTypes from 'prop-types'
import styled from 'react-emotion'
import Title from './Title'

const Div = styled.div`
  color: #fff;
  background-color: #2572b4;
  border-color: #2572b4;
  padding: 10px 15px;
  border-bottom: 1px solid transparent;
  border-top-right-radius: 3px;
  border-top-left-radius: 3px;
`

const Header = ({ title }) => (
  <Div>
    <Title>{title}</Title>
  </Div>
)

Header.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
}

export default Header
