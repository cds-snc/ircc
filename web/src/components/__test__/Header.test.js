import React from 'react'
import { mount, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Header from '../Header'

describe('<Header />', () => {
  it('can be rendered', () => {
    let wrapper = shallow(<Header title="Foo" />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('displays a title supplied by props', () => {
    let wrapper = mount(<Header title="Foo" />)
    expect(wrapper.text()).toEqual('Foo')
  })

  it('displays a title supplied by props', () => {
    let wrapper = mount(<Header title="Foo" />)
    expect(wrapper.text()).toEqual('Foo')
  })

  it('has a blue background', () => {
    let wrapper = shallow(<Header title="Foo" />).dive()
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
