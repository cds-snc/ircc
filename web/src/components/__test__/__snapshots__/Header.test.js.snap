// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> can be rendered 1`] = `
<Styled(div)>
  <Styled(h2)>
    Foo
  </Styled(h2)>
</Styled(div)>
`;

exports[`<Header /> has a blue background 1`] = `
<div
  className="css-x3v3j7"
>
  <Styled(h2)>
    Foo
  </Styled(h2)>
</div>
`;
